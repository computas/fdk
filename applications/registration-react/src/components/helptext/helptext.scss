@import '../../../node_modules/designsystemet/fdk-designsystem-bootstrap4/scss/colors.scss';

.fdk-reg-helptext {
  background-color: $color6;
  border-radius: 5px;
  font-size: 1.4rem;
  strong {
    font-size: 1.8rem;
  }
  .fdk-reg-helptext-more {
    text-decoration: underline;
    white-space: nowrap;
  }
}

.fdk-obligatorisk {
  font-size: 1.1rem;
  color: #ffffff;
  background-color: #4B555F;
  padding: .1em .6em;
  border-radius: 2px;
  position: relative;
  top: -2px;
}