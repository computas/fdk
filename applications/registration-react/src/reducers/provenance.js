import {
  PROVENANCE_REQUEST,
  PROVENANCE_SUCCESS,
  PROVENANCE_FAILURE
} from '../constants/ActionTypes';

export default function provenance(
  state = { isFetchingProvenance: false, provenanceItems: null },
  action
) {
  switch (action.type) {
    case PROVENANCE_REQUEST: {
      return {
        ...state,
        isFetchingProvenance: true
      };
    }
    case PROVENANCE_SUCCESS: {
      const objFromArray = action.response.data.reduce(
        (accumulator, current) => {
          accumulator[current.code] = current; // eslint-disable-line no-param-reassign
          return accumulator;
        },
        {}
      );
      return {
        ...state,
        isFetchingProvenance: false,
        provenanceItems: objFromArray
      };
    }
    case PROVENANCE_FAILURE: {
      return {
        ...state,
        isFetchingProvenance: false,
        provenanceItems: null
      };
    }
    default:
      return state;
  }
}
