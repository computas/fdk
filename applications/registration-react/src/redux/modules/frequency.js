import {
  FREQUENCY_REQUEST,
  FREQUENCY_SUCCESS,
  FREQUENCY_FAILURE
} from '../../constants/ActionTypes';

export default function provenance(
  state = { isFetchingFrequency: false, frequencyItems: null },
  action
) {
  switch (action.type) {
    case FREQUENCY_REQUEST:
      return {
        ...state,
        isFetchingFrequency: true
      };
    case FREQUENCY_SUCCESS: {
      const frequencyItems = action.payload.map(item => ({
        uri: item.uri,
        code: item.code,
        prefLabel_no: item.prefLabel.no,
        prefLabel_nb: item.prefLabel.nb
      }));
      return {
        ...state,
        isFetchingFrequency: false,
        frequencyItems
      };
    }
    case FREQUENCY_FAILURE: {
      return {
        ...state,
        isFetchingFrequency: false,
        frequencyItems: null
      };
    }
    default:
      return state;
  }
}
