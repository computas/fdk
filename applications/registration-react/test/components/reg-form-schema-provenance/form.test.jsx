import React from 'react';
import { shallow } from 'enzyme';
import FormProvenance from '../../../src/components/reg-form-schema-provenance/form';
import helptext from '../../fixtures/helptext';
import frequency from '../../fixtures/frequency';
import provenance from '../../fixtures/provenance';

let defaultProps;
let wrapper;

beforeEach(() => {
  const { helptextItems } = helptext;
  const { frequencyItems } = frequency;
  const { provenanceItems } = provenance;
  defaultProps = {
    initialValues: {
      frequencyItems,
      provenanceItems
    },
    helptextItems,
    provenanceItems,
    input: {
      name: 'provenance',
      value: {
        uri: 'http://data.brreg.no/datakatalog/provinens/bruker',
        code: 'BRUKER',
        prefLabel: {
          nb: 'Brukerinnsamlede data',
          nn: 'Brukerinnsamlede data',
          en: 'User collection'
        }
      }
    }
  };
  wrapper = shallow(<FormProvenance {...defaultProps} />);
});

test('should render FormProvenance correctly', () => {
  expect(wrapper).toMatchSnapshot();
});
