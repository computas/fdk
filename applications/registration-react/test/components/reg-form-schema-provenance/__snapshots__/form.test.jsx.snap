// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render FormProvenance correctly 1`] = `""`;
